### PR TITLE
lua51Packages.nfd: fix module not loading

### LINKED ISSUE #295022: `lua51Packages.nfd` is broken
### Describe the bug
After installing `lua51Packages.nfd`, running `require("nfd")` in lua 5.1 produces the error:
```
clua: error loading module 'nfd' from file '/nix/store/xasw4xnar4b6dq1z2jn1vq9p2rfxhnzz-lua5.1-nfd-scm-1/lib/lua/5.1/nfd.so':
	/nix/store/xasw4xnar4b6dq1z2jn1vq9p2rfxhnzz-lua5.1-nfd-scm-1/lib/lua/5.1/nfd.so: undefined symbol: NFDi_IsFilterSegmentChar
stack traceback:
	[C]: ?
	[C]: in function 'require'
	(command line):1: in main chunk
	[C]: ?
```

### Steps To Reproduce
```shell
nix-shell -E '{ pkgs ? import (fetchTarball https://github.com/NixOS/nixpkgs/tarball/master) {} }: with pkgs; mkShell { packages = with lua51Packages; [ lua nfd ]; }' --run "lua -e 'print(require(\"nfd\").open())'"
```

### Expected behavior
The command above should open a file selection dialog, and print the selected file path.

### Additional context
Changing `master` to `nixos-23.11` gives the expected result, so there is some commit that broke the package. Currently, the commits that changed `pkgs/development/lua-modules/nfd/default.nix` between `nixos-23.11` and `master` are 089f45c and 50e877e, both authored by @teto. I wanted to find out which exact commit caused the breaking, but building from the past nixpkgs failed due to an error in building `tzdata-2024a`, so I do not know which exact commit caused the breaking.

By the way, the version of the `nfd` package is `ncm-1`. Is it right? If the upstream is not tagged with appropriate versions, maybe at least use the commit hash as the version; otherwise, users won't get updates without collecting garbage if you never change the version field in nixpkgs.

### Notify maintainers

@Scoder12

### Metadata
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.78, NixOS, 23.11 (Tapir), 23.11.4575.526d051b128b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.11, nixos-hardware, nixos-unstable"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Petingoso**
As #295022 complains, it wasn't finding symbols properly. 
This PR fixes #295022 .

I traced it to the [migration](https://github.com/NixOS/nixpkgs/pull/288669) from buildLuarocksPackage.extraVariables to buildLuarocksPackage.luarocksConfig.variables where someone forgot to add .variables.

It now launches the dialogue properly.

## Description of changes
Added a missing property

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by NotAShelf — APPROVED** [review]
LGTM!

**Comment by teto** [comment]
could you run the tests to prevent further regressions please ? there seems to be busted tests. If that proves too hard, you could at least require the module. There are examples in lua/overrides.nix

**Comment by Petingoso** [comment]
> could you run the tests to prevent further regressions please ? there seems to be busted tests. If that proves too hard, you could at least require the module. There are examples in lua/overrides.nix

I can't seem to find any appropriate tests under nixos tests, under the pkgs.lua51packages.nfd.passthru.tests nor in pkgs/tests.

I'm not quite sure about what module you're talking about overriding.

I apologize for my lack of experience with all of these frameworks

**Comment by teto** [comment]
sry if that was not clear, my suggestion was to add tests to this nix package because there are none while the source itself seems to have busted tests at least https://github.com/Alloyed/nativefiledialog/tree/master/lua/spec .
You can see how it's done at
https://github.com/NixOS/nixpkgs/blob/73cb87fe1534b9e3a37a6c3e04e7c1001f464044/pkgs/development/lua-modules/overrides.nix#L66
(hopefully it will get easier).
No commit in seven years though.

**Comment by Petingoso** [comment]
> sry if that was not clear, my suggestion was to add tests to this nix package because there are none while the source itself seems to have busted tests at least https://github.com/Alloyed/nativefiledialog/tree/master/lua/spec . You can see how it's done at
> 
> https://github.com/NixOS/nixpkgs/blob/73cb87fe1534b9e3a37a6c3e04e7c1001f464044/pkgs/development/lua-modules/overrides.nix#L66
> 
> (hopefully it will get easier).
> No commit in seven years though.

It actually pulls from a more modern fork, https://github.com/Vexatos/nativefiledialog/tree/master/test.
Issue is I'm trying to go through the different phases via a nixshell but being everything wrapped in buildLuaRocksPackage I'm not understanding how to go about it as at the time of CheckPhase i don't seem to have anything built.

Right now when i trigger a CheckPhase it seems luarocks complains `Argument missing: please specify a rockspec to use on current directory.`

Being the snippet reproducible with just `doCheck = true;` inside buildLuarocksPackage

**Comment by Petingoso** [comment]
After wrestling with this, I think i have tests working

**Comment by UlyssesZh** [comment]
Please add "fix #295022" in the PR comment so that GitHub will automatically close that issue when this PR gets merged.

**Comment by UlyssesZh** [comment]
For small fixing commits like cba74580b, please squash into previous commits. See [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions).

File: pkgs/development/lua-modules/nfd/default.nix [hunk 1]
```diff
@@ -22,15 +21,22 @@ buildLuarocksPackage {
...
 
   buildInputs = lib.optionals stdenv.isDarwin [ AppKit ];
 
-  fixupPhase = ''
+  postInstall = ''
     find $out -name nfd_zenity.so -execdir mv {} nfd.so \;
   '';
 
+  doInstallCheck = true;
+  installCheckInputs = [ lua51Packages.busted ];
```
Comment (teto) [inline pkgs/development/lua-modules/nfd/default.nix:34]: use `lua.pkgs.busted` instead since this will mix a 5.1 version when installing lua_52.pkkgs.nfd

**Comment by teto** [comment]
thank you

**Pull request merged**